src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -360,12 +360,13 @@ class Crystal::Doc::Generator
     filename[@base_dir.size..-1]
   end
 
-  record RelativeLocation, filename : String, line_number : Int32, url : String? do
+  record RelativeLocation, filename : String, line_number : Int32, url : String?, duplicate : Bool do
     def to_json(builder : JSON::Builder)
       builder.object do
         builder.field "filename", filename
         builder.field "line_number", line_number
         builder.field "url", url
+        builder.field "duplicate", duplicate
```

Comment:
I don't think this field needs to be serialized. It doesn't have much meaning outside the immediate template context.
If need be, this can be easily determined by a consuming application.

Reply:
But don't I need to do this so I can use it in the type.html ECR file?

Reply:
No. ECR templates directly access the instance methods of the Crystal object. This `to_json` method is only for generating a JSON export.